.github/pull_request_template.md

```diff
@@ -1,23 +1,34 @@
 resolves #
+docs dbt-labs/docs.getdbt.com/#
 
 <!---
   Include the number of the issue addressed by this PR above if applicable.
   PRs for code changes without an associated issue *will not be merged*.
   See CONTRIBUTING.md for more information.
+
+  Include the number of the docs issue that was opened for this issue. If
+  this change has no user-facing implications, "N/A" suffices instead. New
+  docs tickets can be created[here](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose).
+-->
+
+### Problem
+
+<!---
+  Describe the problem this PR is solving. What is the application state
+  before this PR is merged?
 -->
 
-### Description
+### Solution
 
 <!---
-  Describe the Pull Request here. Add any references and info to help reviewers
-  understand your changes. Include any tradeoffs you considered.
+  Describe the way this PR solves the above problem. Add as much detail as you
+  can to help reviewers understand your changes. Include any alternatives and
+  tradeoffs you considered.
 -->
 
 ### Checklist
 
 - [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
-- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [ ] I have run this code in development and it appears to resolve the stated issue
 - [ ] This PR includes tests, or tests are not required/relevant for this PR
-- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
-- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)
+- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX.
```

Comment:
Is there a reasonable way to indicate which of the two mutually exclusive choices was applicable?

When I see a checked box that has two mutually exclusive items ("A or B"), I wonder -- okay, which one was it?

Reply:
I pushed a new commit that attempts to address this. I can revert it if we don't like it though.

Reply:
I'm finding it difficult to format it nicely in markdown without it looking like a huge checklist now. How important is it for this distinction to be in the checklist to you?

Reply:
Here's the [commit](https://github.com/dbt-labs/dbt-core/blob/7ad6628222af66d0efcd343e6920e018cbd8cbdb/.github/pull_request_template.md) with my best attempt at modeling the split.

Reply:
I'd prefer for it to be clear, especially as it relates to interface changes. Agreed that a short checklist looks a lot better than a long one.

If our template were a little verbose like your attempt, how onerous would it be to edit it down after opening the PR? i.e., like the "Initial" vs. "Edited" below? Or how painful would it be for folks to see the longer checklist?

## Initial

```markdown
### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [ ] I have run this code in development and it appears to resolve the stated issue  
Tests - choose one:  
- [ ] This PR includes tests  
- [ ] Tests are not required or are not relevant for this PR  
Interfaces - choose one:  
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc)  
- [ ] This PR has already received feedback and approval from Product or DX.  
```

## Edited

```markdown
### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] Tests are not required or are not relevant for this PR  
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc)  
```

Reply:
This is a totally reasonable idea, but it is a process change from how the checklist has functioned before 🤔